### ISSUE TITLE
changes object of pullChanges not syncs with database

### ISSUE THREAD

**Issue by sahanjith** [Nozbe/WatermelonDB#1203]
Hi, 

I am experimenting this library to be used in a react native mobile app to sync data with a NodeJS backend (GraphQL endpoint). Synchronization works fine with _pushChanges_  function, but with _pullChanges_ ; even though mobile app received the changes object from backend it does not sync with the relevant local database table.  It gives the warning below.  

`You are trying to sync a collection named 0, but it does not exist. Will skip it (for forward-compatibility). If this is unexpected, perhaps you forgot to add it to your Database constructor's modelClasses property?` 

Object receiving from backend as follows, 

` job_requests: {
          created: [
            {
              id: 'aaaa',
              request_id: 'Foo',
              request_type: 'test',
              job_title: 'aaaa',
              description: 'Foo',
              notes: 'test',
              account_id: '455',
              site_id: null,
              asset_id: null,
              scheduled_at: null,
              earliest_possible_date: null,
              latest_possible_date: null,
              job_status_id: null,
              job_status: 'pending',
              request_flag: 1,
              organization_id: null,
            },
          ],
          updated: [],
          deleted: [],
        },`

Warning refers to the number of characters in change object I guess, as I pass an empty object from backend `{}` it gives the warning 02 times, stating as `collection named 0` &  `collection named 1`. For the changes object above, waring appears for 377 times respectively the number of characters in above object.  

I am using _axios_ to call for backend APIs & my local table schema as follows; 

`  version: 1,
  tables: [
    tableSchema({
      name: 'job_requests',
      columns: [
        { name: 'request_id', type: 'string' },
        { name: 'request_type', type: 'string', isOptional: true },
        { name: 'job_title', type: 'string' },
        { name: 'description', type: 'string', isOptional: true },
        { name: 'notes', type: 'string', isOptional: true },
        { name: 'account_id', type: 'string' },
        { name: 'site_id', type: 'string', isOptional: true },
        { name: 'asset_id', type: 'string', isOptional: true },
        { name: 'scheduled_at', type: 'string', isOptional: true },
        { name: 'earliest_possible_date', type: 'string', isOptional: true },
        { name: 'latest_possible_date', type: 'string', isOptional: true },
        { name: 'job_status_id', type: 'string' },
        { name: 'job_status', type: 'string' },
        { name: 'request_flag', type: 'number' },
        { name: 'organization_id', type: 'string', isOptional: true },
      ],
    }),
  ],` 

Do I have to manually insert the data received from changes objects to local database or Watermelon sync function does that task by itself?

**Comment by sahanjith** [comment]
Issue was that even though I thought I'm receiving an object from _axios_, it was a string. Fixed the issue by parsing the _changes_ object.

**Comment by KrisLau** [comment]
In case anyone else might have ended up here, I ran into a similar issue as OP but I had already parsed the changes using middleware. 

For me, I'm using Objection.js and Knex as my ORM and my problem was caused by using [`knexSnakeCaseMappers`](http://vincit.github.io/objection.js/api/objection/#knexsnakecasemappers) when setting it up. So my `changes` object was returning as camelCase column names instead of snake_case (e.g `firstName` instead of `first_name`) so WatermelonDB did not recognize them.

As soon as I removed the `knexSnakeCaseMappers`, the sync worked as intended.
